### ISSUE TITLE
Charts and plots not rendering with user defined title text_font_size in em

### ISSUE THREAD

**Issue by Rubyj** [bokeh/bokeh#5337]
As the title states, If I add a defined font size to a title on a plot or a chart, that plot and/or chart does not render in Bokeh 0.12.3. This is a pretty large issue. Reproducible below:

```
#!/usr/bin/python
# -*- coding: utf-8 -*-
from bokeh import plotting
from bokeh.models import ColumnDataSource
from bokeh.io import curdoc, show
import pandas as pd

data = [{"1": 10, "2": 4, "3": 3,
         "4": 5, "5": 15, "6": 7,
         "7": 5, "8": 10}]

source = pd.DataFrame(data)
source = ColumnDataSource(source)

fig = plotting.figure(width=1200, height=900, title="BL-55-T55555")

fig.vbar(top="1", x=1, width=0.5, fill_color="blue", line_color="black", source=source)
fig.vbar(top="2", x=2, width=0.5, fill_color="green", line_color="black", source=source)
fig.vbar(top="3", x=3, width=0.5, fill_color="yellow", line_color="black", source=source)
fig.vbar(top="4", x=4, width=0.5, fill_color="orange", line_color="black", source=source)
fig.vbar(top="5", x=5, width=0.5, fill_color="red", line_color="black", source=source)
fig.vbar(top="6", x=6, width=0.5, fill_color="purple", line_color="black", source=source)
fig.vbar(top="7", x=7, width=0.5, fill_color="grey", line_color="black", source=source)
fig.vbar(top="8", x=8, width=0.5, color="black", source=source)

fig.title.text_font_size = '1.9em'

curdoc().add_root(fig)

show(fig)
```

**Comment by bryevdv** [comment]
@Rubyj, the issue is confined only to font sizes expressed in `em`, and there is a simple workaround for now:
```
fig.title.text_font_size = dict(value='1.9em')
```

ping @mattpap seems that `em` needs to be added to this expression? 
```
_font_size_re = re.compile("^[0-9]+(\.[0-9]+)?(ex|px|cm|mm|in|pt|pc)$")
```

**Comment by Rubyj** [comment]
@bryevdv Confirmed the workaround indeed works for now. Thanks.

**Comment by bryevdv** [comment]
@mattpap we should add a test when we fix this.

**Comment by bryevdv** [comment]
@Rubyj this should be fixed in #5347, will try to get a dev build out with it soon

**Comment by Rubyj** [comment]
@bryevdv great thank you for the update. For now, the workaround suffices
